transport/src/test/java/io/netty5/nativeimage/ChannelHandlerMetadataUtil.java

```diff
@@ -0,0 +1,217 @@
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty5.nativeimage;
+
+import com.google.common.reflect.TypeToken;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import io.netty5.channel.ChannelHandler;
+import io.netty5.channel.NativeImageHandlerMetadataTest;
+import org.junit.jupiter.api.Assertions;
+import org.reflections.Reflections;
+import org.reflections.util.ConfigurationBuilder;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.text.Collator;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Generates native-image reflection metadata for subtypes of {@link io.netty5.channel.ChannelHandler}.
+ * <p>
+ * To use, create a JUnit test in the desired Netty module and invoke {@link #generateMetadata(String, String...)} with:
+ * 1. The relative path to the native-image handler reflection metadata file in the Netty module.
+ * This path is relative to the root of the target Netty module.
+ * 2. A list of packages present in the target Netty module that may contain subtypes of the ChannelHandler.
+ * <p>
+ * See {@link NativeImageHandlerMetadataTest}
+ */
+public final class ChannelHandlerMetadataUtil {
+
+    @SuppressWarnings("UnstableApiUsage")
+    private static final Type HANDLER_METADATA_LIST_TYPE = new TypeToken<List<HandlerMetadata>>() {
+    }.getType();
+    private static final Gson gson = new GsonBuilder().setPrettyPrinting().create();
+
+    private ChannelHandlerMetadataUtil() {
+    }
+
+    public static void generateMetadata(String projectRelativeResourcePath, String... packageNames) {
+        Set<Class<? extends ChannelHandler>> subtypes = findChannelHandlerSubclasses(packageNames);
+
+        if (Arrays.asList(packageNames).contains("io.netty5.channel")) {
+            // We want the metadata for the ChannelHandler itself too
+            subtypes.add(ChannelHandler.class);
+        }
+
+        Set<HandlerMetadata> handlerMetadata = subtypes.stream()
+                .map(type -> new HandlerMetadata(type.getName(), new Condition(type.getName()), true))
+                .collect(Collectors.toSet());
+
+        Path existingMetadataPath = new File(projectRelativeResourcePath).toPath().toAbsolutePath();
```

Comment:
Maven seems to always set the current directory to the module it's currently building/testing - this lets us look up the `reflect-config.json` file directly from the module sources by using a relative path here